### PR TITLE
[candi] Parallel download registry packages in separate step before installation

### PR DESCRIPTION
**Pull request by asviel**
## Description
Change the behavior of installing packages from registry (`bb-rp-install`) to pre-download in separate bashible step followed by installation from local cache.

- Now `d8-curl` is used with parallel downloading of multiple packages.

**UPD:** During e2e tests we found bugs in certificate handling in the AltLinux bundle. They were fixed in the same PR.

## Why do we need it, and what problem does it solve?
Now we can separate package installation and fetching. It can be useful with slow registries. If there are problems with downloading packages from the registry, the bashible will "hang" at the beginning rather than throwing an error somewhere in the middle.

## Why do we need it in the patch release (if we do)?
We don't

## What is the expected result?
`bb-rp-install` install package from local temp storage and run `bb-rp-fetch` if package not found locally. In other words, the function's result should not change.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: Parallel download registry packages in separate step before installation.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/gcp'...

**Comment by asviel** [comment]
/e2e/abort static;Static;containerd;1.25 6810474211 6810474211-3-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.252

**Comment by asviel** [comment]
/e2e/abort static;Static;containerd;1.25 6814541192 6814541192-2-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.14

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/static'...

**Comment by asviel** [comment]
/e2e/abort static;Static;containerd;1.25 6810474211 6810474211-3-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.252

**Comment by asviel** [comment]
/e2e/abort static;Static;containerd;1.25 6814541192 6814541192-2-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.14

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.25 6814541192 6814541192-1-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.7

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.25 6821996185 6821996185-1-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.84

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.25 6814541192 6814541192-1-con-1-25 /sys/deckhouse-oss/install:pr6415 astra@192.168.199.7

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6810473805 6810473805-3-con-1-25 /sys/deckhouse-oss/install:pr6415 centos@18.156.141.226

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6810473805 6810473805-1-con-1-25 /sys/deckhouse-oss/install:pr6415 centos@3.124.69.130

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6810473805 6810473805-2-con-1-25 /sys/deckhouse-oss/install:pr6415 centos@3.69.93.164

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6814540604 6814540604-1-con-1-25 /sys/deckhouse-oss/install:pr6415 centos@35.157.241.177

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6810473848 6810473848-1-con-1-25 /sys/deckhouse-oss/install:pr6415 cloud-user@158.160.117.59

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

File: candi/bashible/bashbooster/50_deckhouse_registrypackages.sh [hunk 1]
```diff
@@ -12,8 +12,19 @@
...
-# shellcheck disable=SC2153
+bb-var BB_RP_FETCHED_PACKAGES_STORE "${TMPDIR}/registrypackages"
+
+# Use d8-curl if installed, fallback to system package if not
+bb-rp-curl() {
+  if command -v d8-curl > /dev/null ; then
+    d8-curl --parallel "$@"
+  else
+    curl "$@"
+  fi
```
Comment (RomanenkoDenys) [inline candi/bashible/bashbooster/50_deckhouse_registrypackages.sh:26]: Please add one more check if default curl supports parallel downloads

Comment (asviel) [reply candi/bashible/bashbooster/50_deckhouse_registrypackages.sh:26]: It's not necessary. `curl` is used only once to download the `d8-curl` and `jq` packages. It doesn't make much sense to download two lightweight packages in parallel.

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by name212 — APPROVED** [review]

**Pull request merged**